### PR TITLE
Normals

### PR DESCRIPTION
**Pull request by seepa**
This PR implements face and vertex normal rendering in UMVE. I have refactored shader loading to support vert/geom/frag shaders with different basenames and implemented two geom shaders for vertex/face normal generation as well as one simple vert and frag shader. The length of the normals is calculated with a 2^x function and the default size looks good for meshes in the unit cube.

### CODE REVIEW

File: libs/ogl/shader_program.cc [hunk 1]
```diff
@@ -101,18 +101,27 @@ ShaderProgram::compile_shader (GLuint shader_id, std::string const& code)
...
-    std::string geom_filename = basename + ".geom";
-    std::string frag_filename = basename + ".frag";
+    return try_load_specific(basename, basename, basename);
+}
+
+bool
+ShaderProgram::try_load_specific (std::string const& vert_basename,
+    std::string const& geom_basename, std::string const& frag_basename)
+{
+    std::string vert_filename = vert_basename + ".vert";
```
Comment (simonfuhrmann) [inline libs/ogl/shader_program.cc]: For this method, we should not append the extension. I.e., construct the filename from the basename in `try_load_all`, and rename the arguments to `vert_filename`, etc, here.

**Comment by vikasTmz** [comment]
@seepa I'm getting `undefined reference` ERROR at  `this->normals_face_shader->try_load_specific()`.

**Comment by seepa** [comment]
yes, I'm getting the same. I'll try to find out why this suddenly doesn't work anymore.

**Comment by seepa** [comment]
@vikasTmz could you try again, please?

I was able to track down the bug with glGetProgramInfoLog(). There was an error when linking vertex and geometry shader: the vertex shader had a vec4 color as output while the geom shader expected a vec3 color as input. For some reason, my PC at work did not complain about this mistake (OpenGL 3.3). I have now removed the color completely, as it is set in the fragment shader manually.

**Comment by simonfuhrmann** [comment]
Closing old PRs.

**Pull request closed without merging**